crates/turborepo-lib/src/config/turbo_config.rs

```diff
@@ -0,0 +1,592 @@
+use std::{collections::HashMap, ffi::OsString};
+
+use anyhow::anyhow;
+use dirs_next::config_dir;
+use serde::{Deserialize, Serialize};
+use turbopath::AbsoluteSystemPathBuf;
+
+use crate::{
+    commands::CommandBase,
+    config::{Error as ConfigError, RawTurboJSON},
+    package_json::PackageJson,
+};
+
+const DEFAULT_API_URL: &str = "https://vercel.com/api";
+const DEFAULT_LOGIN_URL: &str = "https://vercel.com";
+const DEFAULT_TIMEOUT: u64 = 20;
+
+macro_rules! create_builder {
+    ($func_name:ident, $property_name:ident, $type:ty) => {
+        pub fn $func_name(mut self, value: $type) -> Self {
+            self.override_config.$property_name = value;
+            self
+        }
+    };
+}
+
+#[derive(Serialize, Deserialize, Default, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ConfigurationOptions {
+    #[serde(alias = "apiurl")]
+    #[serde(alias = "ApiUrl")]
+    #[serde(alias = "APIURL")]
+    pub(crate) api_url: Option<String>,
+    #[serde(alias = "loginurl")]
+    #[serde(alias = "LoginUrl")]
+    #[serde(alias = "LOGINURL")]
+    pub(crate) login_url: Option<String>,
+    #[serde(alias = "teamslug")]
+    #[serde(alias = "TeamSlug")]
+    #[serde(alias = "TEAMSLUG")]
+    pub(crate) team_slug: Option<String>,
+    #[serde(alias = "teamid")]
+    #[serde(alias = "TeamId")]
+    #[serde(alias = "TEAMID")]
+    pub(crate) team_id: Option<String>,
+    pub(crate) token: Option<String>,
```

Comment:
Can you verify that this won't read a token from `turbo.json`, which we expect to be checked in?

Reply:
Yeah, I'll make sure there is a test for it. Some thoughts about "checkin-viable tokens" over here:
https://www.notion.so/Turborepo-Configuration-a15dd51ca99c4f56983fc38d86784035?d=ab20442220954f4ab635ab98d7bbcf28&pvs=4#e953df8e99f448df8b282f361b959f8c